services/blockchain-indexer/shared/init.js

```diff
@@ -0,0 +1,51 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2023 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const {
+	LoggerConfig,
+	Logger,
+} = require('lisk-service-framework');
+
+const config = require('../config');
+
+LoggerConfig(config.log);
+
+const { getPosConstants } = require('./dataService/business/pos/constants');
+const status = require('./indexer/indexStatus');
+const processor = require('./processor');
+
+const logger = Logger();
+
+const init = async () => {
+	try {
+		await getPosConstants();
+
+		// Init database
+		await status.init();
+
+		if (config.operations.isIndexingModeEnabled) {
+			await processor.init();
+		}
```

Comment:
Could you please add the logic/invocation to initialize the tables below?
Probably we can add any other necessary module constants caching invocations here?
```suggestion
		// Initialize database
		// Add logic here
		
		// Update the constants cache
		await getPosConstants();

		// Initialize index status updates
		await indexStatus.init();

		if (config.operations.isIndexingModeEnabled) {
			await processor.init();
		}
```

Reply:
We need to explicitly initialize the DB even if the indexing mode is not enabled. The tables should be available before any queries to the DB is made by the `reader` user in case we use the dataService mode only.